coral-common/src/main/java/com/linkedin/coral/common/CoralJavaTypeFactoryImpl.java

```diff
@@ -0,0 +1,25 @@
+/**
+ * Copyright 2023 LinkedIn Corporation. All rights reserved.
+ * Licensed under the BSD-2 Clause license.
+ * See LICENSE in the project root for license information.
+ */
+package com.linkedin.coral.common;
+
+import java.util.List;
+
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.rel.type.RelDataType;
+import org.apache.calcite.rel.type.RelDataTypeSystem;
+import org.apache.calcite.rel.type.StructKind;
+
+
+public class CoralJavaTypeFactoryImpl extends JavaTypeFactoryImpl {
```

Comment:
Please add java doc to document why the override is being introduced.

Reply:
Done